### ISSUE TITLE
Histogram bucket increase shows fractions in value

### ISSUE THREAD

**Issue by TLS566ba** [prometheus/prometheus#13156]
### What did you do?

I created a [histogram metric](https://prometheus.io/docs/concepts/metric_types/#histogram) in python using the official [python client](https://prometheus.github.io/client_python/instrumenting/histogram/):

```python
my_hist_metric = Histogram(
        "epoch_duration_seconds_bucket",
        "Duration of one machine learning training epoch.",
        buckets=(0.1, 0.5, 1.0, 2.5, 5.0, 7.5, 10.0, 12.5, 15.0, 17.5, 20.0, 25.0, 30.0, 60.0, 120.0, 240.0, 600.0, INF)
    )
```
The global `scrape_interval` was set to `30s`.

After a few hours, I sent the PromQL query
```promql
sum(increase(epoch_duration_seconds_bucket[2m])) by (le)
```

### What did you expect to see?

Integer values of the individual buckets, since the values denote the `increase` of each bucket, so this should be the SI unit "number" without any fractional part (unlike `rate`).

### What did you see instead? Under which circumstances?

Floating point values with fractional parts, such as, e.g., `25.3` or `26.7`. See screenshot of query result table:

![image](https://github.com/prometheus/prometheus/assets/144460191/04d9aba4-2d92-4b8b-81e6-9e23c6bd17ac)

### System information

Azure Monitor managed service for Prometheus

### Prometheus version

```text
I could not find this in Azure Portal. Apart from the GoVersion, which is `go1.19.13`, the build information table is empty. Seems as Microsoft does not provide that kind of information.
```

**Comment by rgroothuijsen** [comment]
According to [the documentation](https://prometheus.io/docs/prometheus/latest/querying/functions/#increase), this appears to be an expected behavior:

> The increase is extrapolated to cover the full time range as specified in the range vector selector, so that it is possible to get a non-integer result even if a counter increases only by integer increments.

**Comment by beorn7** [comment]
Indeed, `increase` and `rate` are doing the same thing, just that `rate` divides by the length of the used range selector.
